src/src/PluginStructs/P035_data_struct.cpp

```diff
@@ -0,0 +1,661 @@
+      stateSize = std::max(stateSize,
+                           (uint16_t)(kFujitsuAcStateLengthShort - 1));
+
+      // If we think it isn't a "short" message.
+      if (stateSize > kFujitsuAcStateLengthShort) {
+        // Then it has to be at least the smaller version of the "normal" size.
+        stateSize = std::max(stateSize, (uint16_t)(kFujitsuAcStateLength - 1));
+      }
+
+      // Lastly, it should never exceed the maximum "normal" size.
+      stateSize = std::min(stateSize, kFujitsuAcStateLength);
+      break;
+    case HITACHI_AC3:
+      // HitachiAc3 has two distinct & different size states, so make a best
+      // guess which one we are being presented with based on the number of
+      // hexadecimal digits provided. i.e. Zero-pad if you need to to get
+      // the correct length/uint8_t size.
+      stateSize = inputLength / 2; // Every two hex chars is a uint8_t.
+      // Use at least the minimum size.
+      stateSize = std::max(stateSize,
+                           (uint16_t)(kHitachiAc3MinStateLength));
+
+      // If we think it isn't a "short" message.
+      if (stateSize > kHitachiAc3MinStateLength) {
+        // Then it probably the "normal" size.
+        stateSize = std::max(stateSize,
+                             (uint16_t)(kHitachiAc3StateLength));
+      }
+
+      // Lastly, it should never exceed the maximum "normal" size.
+      stateSize = std::min(stateSize, kHitachiAc3StateLength);
+      break;
+    case MWM:
+      // MWM has variable size states, so make a best guess
+      // which one we are being presented with based on the number of
+      // hexadecimal digits provided. i.e. Zero-pad if you need to to get
+      // the correct length/uint8_t size.
+      stateSize = inputLength / 2; // Every two hex chars is a uint8_t.
+      // Use at least the minimum size.
+      stateSize = std::max(stateSize, (uint16_t)3);
+
+      // Cap the maximum size.
+      stateSize = std::min(stateSize, kStateSizeMax);
+      break;
+    case SAMSUNG_AC:
+      // Samsung has two distinct & different size states, so make a best guess
+      // which one we are being presented with based on the number of
+      // hexadecimal digits provided. i.e. Zero-pad if you need to to get
+      // the correct length/uint8_t size.
+      stateSize = inputLength / 2; // Every two hex chars is a uint8_t.
+      // Use at least the minimum size.
+      stateSize = std::max(stateSize, (uint16_t)(kSamsungAcStateLength));
+
+      // If we think it isn't a "normal" message.
+      if (stateSize > kSamsungAcStateLength) {
+        // Then it probably the extended size.
+        stateSize = std::max(stateSize,
+                             (uint16_t)(kSamsungAcExtendedStateLength));
+      }
+
+      // Lastly, it should never exceed the maximum "extended" size.
+      stateSize = std::min(stateSize, kSamsungAcExtendedStateLength);
+      break;
+    default: // Everything else.
+      stateSize = IRsend::defaultBits(irType) / 8;
```

Comment:
Just curious; Are there protocols with a bit length not a multiple of 8?
Not sure what kind of bugs we might run into for them, but I can imagine it may cause some out-of-bound errors if used to reserve/allocate some array.
A quick fix is then (if it is a problem): `stateSize = (IRsend::defaultBits(irType) + 7) / 8;`

Reply:
Don't expect that to be a problem, as this plugin has been used quite long and thus tested extensively. I can apply that fix though. And `stateSize` is of type `uint16_t`.

Reply:
Not sure if it will ever be a problem, I just noticed it while quickly browsing through the code.